xds/internal/balancer/clusterresolver/balancerconfig/configbuilder.go

```diff
@@ -176,6 +170,9 @@ func buildClusterImplConfigForEDS(parentPriority int, edsResp xdsclient.Endpoint
 	}
 
 	priorityChildNames, priorities := groupLocalitiesByPriority(edsResp.Localities)
+	retNames := make([]string, 0, len(priorityChildNames))
+	retAddrs := make([]resolver.Address, 0, len(priorityChildNames))
```

Comment:
We could get more than one entry added to `retAddrs` per iteration of the `for` loop. So, we should revert this change.

Reply:
In this case we're allocating at least the minimal capacity, I think it's better than starting from zero, because it grows exponentially 1,2,4,8 it does helps with small `priorityChildNames` numbers.

Reply:
Ack.